LukasWaveformReaderApp/Db/waveform.template

```diff
@@ -174,88 +174,123 @@ record(ao,      "$(P):Z:OFFSET:END:${channel}") {
   field(DTYP, "asynFloat64ArrayIn")
   field(FTVL,    "DOUBLE")
   field(NELM,    "$(WAVEFORM_SIZE)")
   field(SCAN, "I/O Intr")
 }
 
-# Values of the waveform before extraction
-record(waveform, "$(P):COMPLETE:WF:${channel}") 
+# Beam loss values, before extracting region of interest
+record(waveform, "$(P):COMPLETE_BEAM_LOSS_${channel}") 
 {
   field(PINI, "1")
-  field(DESC,    "Subarray")
-  field(INP, "@asyn($(PORT),$(ADDR),$(TIMEOUT))COMPLETE_WAVEFORM:$(channel)")
+  field(DESC,    "Beam loss values")
+  field(INP, "@asyn($(PORT),$(ADDR),$(TIMEOUT))COMPLETE_BEAM_LOSS_$(channel)")
   field(DTYP, "asynInt16ArrayIn")
   field(FTVL,    "SHORT")
   # NELM is the maximum number of values, NORD is the number of values actually read
   field(NELM,    "$(WAVEFORM_SIZE)")
   field(SCAN, "I/O Intr")
 }
 
-# The X-axis of the final extracted waveform, contains distance values in m
-record(waveform, "$(P):EXTRACTED:X:WF:${channel}") 
+# The Z-axis of the region of interest, contains distance values in m
+record(waveform, "$(P):ROI_Z_${channel}") 
 {
   field(PINI, "1")
-  field(DESC,    "X Subarray")
-  field(INP, "@asyn($(PORT),$(ADDR),$(TIMEOUT))EXTRACTED_X_AXIS:$(channel)")
+  field(DESC,    "Z values of ROI")
+  field(INP, "@asyn($(PORT),$(ADDR),$(TIMEOUT))ROI_Z_$(channel)")
   field(DTYP, "asynFloat64ArrayIn")
   field(FTVL,    "DOUBLE")
   field(NELM,    "$(WAVEFORM_SIZE)")
   field(SCAN, "I/O Intr")
 }
 
-# Values of the extracted waveform
-record(waveform, "$(P):EXTRACTED:WF:${channel}") 
+# Beam loss values in ROI
+record(waveform, "$(P):ROI_BEAM_LOSS_${channel}") 
 {
   field(PINI, "1")
-  field(DESC,    "Extracted Subarray")
-  field(INP, "@asyn($(PORT),$(ADDR),$(TIMEOUT))EXTRACTED_WAVEFORM:$(channel)")
+  field(DESC,    "Beam loss values in ROI")
+  field(INP, "@asyn($(PORT),$(ADDR),$(TIMEOUT))ROI_BEAM_LOSS_$(channel)")
   field(DTYP, "asynInt16ArrayIn")
   field(FTVL,    "SHORT")
   # NELM is the maximum number of values, NORD is the number of values actually read
   field(NELM,    "$(WAVEFORM_SIZE)")
   field(SCAN, "I/O Intr")
-}
\ No newline at end of file
+}
+
+# Store the time passed between the current and previous 
+# retrieval of data from a stream
+record(ai, "$(P):INTERVAL_${channel}")
+{
+    field(DTYP, "asynInt32")
+    #field(PINI, "YES")
+    field(INP, "@asyn($(PORT),$(ADDR),$(TIMEOUT))INTERVAL${channel}")
+    field(SCAN, "I/O Intr")
+    #field(VAL, "100")
+}
+
+# Set to TRUE if the waveform should be extracted, FALSE if not
+# determined by whether the max value of the waveform data is
+# above the threshold or not
+record(bi, "$(P):EXTRACT_RBV_${channel}")
+{
+    field(SCAN, "I/O Intr")
+    field(DESC, "Whether to extract waveform")
+    field(DTYP, "asynUInt32Digital")
+    field(INP, "@asynMask($(PORT),$(ADDR), 1, $(TIMEOUT))EXTRACT${channel}")
+    field(ZNAM, "FALSE")
+    field(ONAM, "TRUE")
+  }
+
+# Used to reset the value of "$(P):EXTRACT_${channel}" from the IOC
```

Comment:
This comment doesn't really make sense? This record seems a little funky, could u explain it's purpose in more detail